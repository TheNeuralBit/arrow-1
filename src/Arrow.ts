import { Table } from './table';
import { readBuffers } from './reader/arrow';
export { Table, readBuffers };

/* These exports are needed for the closure umd targets */
try {
    const Arrow = eval('exports');
    if (typeof Arrow === 'object') {
        // string indexers tell closure compiler not to rename these properties
        Arrow['Table'] = Table;
        Arrow['readBuffers'] = readBuffers;
    }
} catch (e) { /* not the UMD bundle */ }
/** end closure exports */
