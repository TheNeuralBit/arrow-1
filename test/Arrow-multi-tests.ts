import * as fs from 'fs';
import * as path from 'path';
import { Table, readBuffers } from './Arrow';

const arrowFolderNames = ['count', 'latlong', 'origins'];
const arrowTestConfigurations = arrowFolderNames.reduce((configs, folder) => {
    const schemaPath = path.resolve(__dirname, `./arrows/multi/${folder}/schema.arrow`);
    const recordsPath = path.resolve(__dirname, `./arrows/multi/${folder}/records.arrow`);
    return [...configs, [folder, fs.readFileSync(schemaPath), fs.readFileSync(recordsPath)]];
}, []);

for (let [folder, schema, records] of arrowTestConfigurations) {
    describe(`multi ${folder}`, () => {
        test(`enumerates vectors`, () => {
            expect.hasAssertions();
            for (let vectors of readBuffers(schema, records)) {
                for (let vector of vectors) {
                    expect(vector.name).toMatchSnapshot();
                    expect(vector.type).toMatchSnapshot();
                    expect(vector.length).toMatchSnapshot();
                        for (let i = -1, n = vector.length; ++i < n;) {
                        expect(vector.get(i)).toMatchSnapshot();
                    }
                }
            }
        });
        test(`concats vectors into a table`, () => {
            expect.hasAssertions();
            const table = Table.from(schema, records);
            for (const vector of table) {
                expect(vector.name).toMatchSnapshot();
                expect(vector.type).toMatchSnapshot();
                expect(vector.length).toMatchSnapshot();
                for (let i = -1, n = vector.length; ++i < n;) {
                    expect(vector.get(i)).toMatchSnapshot();
                }
            }
        });
    });
}