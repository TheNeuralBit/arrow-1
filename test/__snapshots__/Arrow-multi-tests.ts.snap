// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multi count concats vectors into a table 1`] = `"row_count"`;

exports[`multi count concats vectors into a table 2`] = `"Int"`;

exports[`multi count concats vectors into a table 3`] = `1`;

exports[`multi count concats vectors into a table 4`] = `10000`;

exports[`multi count enumerates vectors 1`] = `"row_count"`;

exports[`multi count enumerates vectors 2`] = `"Int"`;

exports[`multi count enumerates vectors 3`] = `1`;

exports[`multi count enumerates vectors 4`] = `10000`;

exports[`multi latlong concats vectors into a table 1`] = `"origin_lat"`;

exports[`multi latlong concats vectors into a table 2`] = `"FloatingPoint"`;

exports[`multi latlong concats vectors into a table 3`] = `5`;

exports[`multi latlong concats vectors into a table 4`] = `35.393089294433594`;

exports[`multi latlong concats vectors into a table 5`] = `35.393089294433594`;

exports[`multi latlong concats vectors into a table 6`] = `35.393089294433594`;

exports[`multi latlong concats vectors into a table 7`] = `29.533695220947266`;

exports[`multi latlong concats vectors into a table 8`] = `29.533695220947266`;

exports[`multi latlong concats vectors into a table 9`] = `"origin_lon"`;

exports[`multi latlong concats vectors into a table 10`] = `"FloatingPoint"`;

exports[`multi latlong concats vectors into a table 11`] = `5`;

exports[`multi latlong concats vectors into a table 12`] = `-97.6007308959961`;

exports[`multi latlong concats vectors into a table 13`] = `-97.6007308959961`;

exports[`multi latlong concats vectors into a table 14`] = `-97.6007308959961`;

exports[`multi latlong concats vectors into a table 15`] = `-98.46977996826172`;

exports[`multi latlong concats vectors into a table 16`] = `-98.46977996826172`;

exports[`multi latlong enumerates vectors 1`] = `"origin_lat"`;

exports[`multi latlong enumerates vectors 2`] = `"FloatingPoint"`;

exports[`multi latlong enumerates vectors 3`] = `5`;

exports[`multi latlong enumerates vectors 4`] = `35.393089294433594`;

exports[`multi latlong enumerates vectors 5`] = `35.393089294433594`;

exports[`multi latlong enumerates vectors 6`] = `35.393089294433594`;

exports[`multi latlong enumerates vectors 7`] = `29.533695220947266`;

exports[`multi latlong enumerates vectors 8`] = `29.533695220947266`;

exports[`multi latlong enumerates vectors 9`] = `"origin_lon"`;

exports[`multi latlong enumerates vectors 10`] = `"FloatingPoint"`;

exports[`multi latlong enumerates vectors 11`] = `5`;

exports[`multi latlong enumerates vectors 12`] = `-97.6007308959961`;

exports[`multi latlong enumerates vectors 13`] = `-97.6007308959961`;

exports[`multi latlong enumerates vectors 14`] = `-97.6007308959961`;

exports[`multi latlong enumerates vectors 15`] = `-98.46977996826172`;

exports[`multi latlong enumerates vectors 16`] = `-98.46977996826172`;

exports[`multi origins concats vectors into a table 1`] = `"origin_city"`;

exports[`multi origins concats vectors into a table 2`] = `"Utf8"`;

exports[`multi origins concats vectors into a table 3`] = `5`;

exports[`multi origins concats vectors into a table 4`] = `"Oklahoma City"`;

exports[`multi origins concats vectors into a table 5`] = `"Oklahoma City"`;

exports[`multi origins concats vectors into a table 6`] = `"Oklahoma City"`;

exports[`multi origins concats vectors into a table 7`] = `"San Antonio"`;

exports[`multi origins concats vectors into a table 8`] = `"San Antonio"`;

exports[`multi origins enumerates vectors 1`] = `"origin_city"`;

exports[`multi origins enumerates vectors 2`] = `"Utf8"`;

exports[`multi origins enumerates vectors 3`] = `5`;

exports[`multi origins enumerates vectors 4`] = `"Oklahoma City"`;

exports[`multi origins enumerates vectors 5`] = `"Oklahoma City"`;

exports[`multi origins enumerates vectors 6`] = `"Oklahoma City"`;

exports[`multi origins enumerates vectors 7`] = `"San Antonio"`;

exports[`multi origins enumerates vectors 8`] = `"San Antonio"`;
